### PR TITLE
Add section on wrappers to overview pages

### LINKED ISSUE #2079: Discoverability of the wrappers
Thank you for a very helpful package!
A few suggestions to improving the discoverability of the wrappers - 

After reading this documentation section

https://github.com/Lightning-AI/torchmetrics/blob/99d6d9d6ac4eb1b3398241df558604e70521e6b0/docs/source/pages/lightning.rst?plain=1#L52-L58

in the [TorchMetrics in PyTorch Lightning](https://torchmetrics.readthedocs.io/en/stable/pages/lightning.html) section, I was left feeling like I had to deal with logging tensors manually. 

However, in some cases, such as multiclass metrics with `average=None`, the [Classwise wrapper](https://torchmetrics.readthedocs.io/en/stable/wrappers/classwise_wrapper.html) can come in handy!

Suggestion for the docs - add a reference to this wrapper, and maybe to others, from the TorchMetrics in PyTorch Lightning documentation page. 

In general, I didn't see any reference to the wrappers (including from the Structure Overview page), which I think is a shame as they can be very useful. 
On this matter the MetricCollection seems similar to the other wrappers, so maybe reference the other wrappers from [that section?](https://torchmetrics.readthedocs.io/en/stable/pages/overview.html#metriccollection)

Also, in the side menu, the Wrappers section seems like just another class of metrics, but that is not the case. Suggestion - rename the section to Metric Wrappers or something like that to distinguish it from the metrics.

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2079

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2085.org.readthedocs.build/en/2085/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by stancld — APPROVED** [review]

**Review by Borda — COMMENTED** [review]
@SkafteNicki mind checking: ` UserWarning: The ``compute`` method of metric SumMetric was called before the ``update`` method which may lead to errors, as metric states have not yet been updated.`

File: docs/source/pages/overview.rst [hunk 1]
```diff
@@ -361,6 +361,52 @@ information on this topic.
...
+    wrapped_metric = ClasswiseWrapper(base_metric, labels=["cat", "dog", "fish"])
+    target = torch.tensor([0, 2, 0, 2, 0, 1, 0, 2])
+    preds = torch.tensor([2, 1, 2, 0, 1, 2, 2, 2])
+    print(base_metric(preds, target))  # this returns a simple tensor without label info
+    print(wrapped_metric(preds, target))  # this returns a dict with label info
+
+.. testoutput::
+    :options: +NORMALIZE_WHITESPACE
+
+    torch.tensor([0.0000, 0.0000, 0.3333])
```
Comment (Borda) [inline docs/source/pages/overview.rst:388]:
```suggestion
    tensor([0.0000, 0.0000, 0.3333])
```

File: docs/source/pages/overview.rst [hunk 2]
```diff
@@ -361,6 +361,52 @@ information on this topic.
...
+     'multiclassaccuracy_dog': tensor(0.),
+     'multiclassaccuracy_fish': tensor(0.3333)}
+
+Another good example of wrappers is the :class:`~torchmetrics.wrappers.BootStrapper` that allows for easy bootstrapping
+of metrics e.g. computation of confidence intervals by resampling of input data.
+
+.. testcode::
+
+    from torchmetrics.classification import MulticlassAccuracy
+    from torchmetrics.wrappers import ClasswiseWrapper
```
Comment (Borda) [inline docs/source/pages/overview.rst:399]:
```suggestion
    from torchmetrics.wrappers import BootStrapper
```
